packages/app/src/vis-packs/core/matrix/utils.ts

```diff
@@ -34,6 +35,22 @@ export function createNumericFormatter(
   }
 }
 
+export function createBigIntFormatter(
+  notation: Notation,
+): (val: ScalarValue<NumericType>) => string {
+  const formatter = createNumericFormatter(notation);
+
+  switch (notation) {
+    case Notation.Scientific:
+      return (val) => formatter(Number(val));
+
+    default:
+      return (val) => {
+        return typeof val === 'bigint' ? val.toString() : formatter(val);
```

Comment:
I've pushed a bug fix here. Int-64 values provided as `number` were being cast to string instead of being formatted with D3. I'm not 100% which is the result we want, actually - e.g. for fixed-point notation, do we want `1.000` or `1`? I assumed `1`, since decimals don't matter, so maybe I should revert and keep the `toString()`?

Reply:
Yeah, it does not make much sense to do more than converting to strings for integers. `toString` should be enough.

Reply:
Refactoring done in https://github.com/silx-kit/h5web/pull/1745/commits/6852665dbbcbe77084896fa6c12cfc57f30e9bf8

I've split `createNumericFormatter` into two:

- `createFloatFormatter` which works as before
- `createIntegerFormatter`:
  - with "auto" and "fixed-point" notation: casts numbers/bigints to string with `toString()`
  - with "scientific" notation: casts bigints to numbers with `Number`, and then formats numbers with D3 (same format as before: `'.3e'`)